### PR TITLE
README.md: Fix (mostly minor) markdown linting issues, etc.

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [X] This is a documentation change only

Issues:
- N/A

Purpose:
- What is this pull request trying to do? Fix some minor issues that VSCode's markdown linter caught
- What release is this for? 0.9.x
- Is there a project or milestone we should apply this to? 0.9.x

### CODE REVIEW

**Review by BenjamenMeyer — COMMENTED** [review]
Q: does it make sense to just change this to RST?

**Comment by stephengtuggy** [comment]
> Q: does it make sense to just change this to RST?

@BenjamenMeyer I'm not sure I follow why RST would be better. I slightly prefer the Markdown format, myself.

**Comment by BenjamenMeyer** [comment]
Most of the changes here would go away, and there's some things in there, like the code examples, that are still RST format not MD format.

**Comment by stephengtuggy** [comment]
@BenjamenMeyer I see. Okay. Perhaps we could change to RST format then.

My only concern would be, if anyone is linking explicitly to README.md rather than to the project root, they will get a broken link after the switchover.

**Comment by stephengtuggy** [comment]
I still think this PR is worth merging for a few of the other, editorial-type fixes

File: README.md [hunk 1]
```diff
@@ -271,16 +270,16 @@ Compiling On Linux
...
    cmake ../engine
-   make -j $(nproc) # (where $(nproc) returns the number of available CPU threads/cores on the system)
+   cmake --build ./build -j $(getconf _NPROCESSORS_ONLN) # (where the getconf clause returns the number of available CPU threads/cores on the system)
    mkdir ../bin && cp vegastrike-engine ../bin/ && cp setup/vegasettings ../bin/ && cd ..
    ```
 
    __TIPS__:
 
-   To enable verbose output for debugging purposes (will show compilation commands), pass the `VERBOSE=1` argument:
+   To enable verbose output for debugging purposes (will show compilation commands), pass the `--version` argument, where supported:
```
Comment (P-D-E) [inline README.md:279]: `--version` here should be `--verbose`, right?

Comment (stephengtuggy) [reply README.md:279]: @P-D-E Oops, you're right. Let me fix that

Comment (stephengtuggy) [reply README.md]: Fixed

**Review by nabaco — APPROVED** [review]

**Pull request merged**